### PR TITLE
[CES-90] - Storage account module

### PR DESCRIPTION
**Pull request by christian-calabrese**
This PR introduces a new Terraform module for creating and managing Azure Storage Accounts. The module provides a comprehensive and flexible way to deploy storage accounts with various configuration options, security features, and performance settings.

Key features of this module include:
1. Configurable environment settings for consistent naming and tagging
2. Support for different tiers (small and large) to accommodate various workloads
3. Customer-managed key integration for enhanced security
4. Network rule configuration for access control
5. Private endpoint and DNS zone integration
6. Static website hosting capabilities
7. Custom domain support
8. Comprehensive blob features including:
   - Immutability policies
   - Versioning
   - Soft delete and restore policies
   - Change feed
   - Last access time tracking

The module adheres to best practices for Azure Storage Account deployment and includes several validation rules to ensure proper configuration.

Testing:
- The module has been tested with various input combinations

### Type of changes

- [ ] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [ ] No

### CODE REVIEW

**Review by mamu0 — APPROVED** [review]
Great job!

File: infra/modules/azure_storage_account/alerts.tf [hunk 1]
```diff
@@ -0,0 +1,35 @@
+# -----------------------------------------------
+# Alerts
+# -----------------------------------------------
+
+resource "azurerm_monitor_metric_alert" "storage_account_health_check" {
+  count               = local.tier_features.advanced_threat_protection ? 1 : 0
+  name                = "[${replace("${var.environment.domain != null ? "${var.environment.domain} | " : ""}${azurerm_storage_account.this.name}", "-", "")}] Low Availability"
```
Comment (Krusty93) [inline infra/modules/azure_storage_account/alerts.tf:7]: Should we use the domain name here? I'm sticking to use the resource name only - just to know

File: infra/modules/azure_storage_account/outputs.tf [hunk 2]
```diff
@@ -0,0 +1,11 @@
+output "name" {
```
Comment (Krusty93) [inline infra/modules/azure_storage_account/outputs.tf:1]: I suggest to add `id` as well

File: infra/modules/azure_storage_account/variables.tf [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    error_message = "Delete retention days must be 0 to disable the policy or between 1 and 365."
+  }
+
+  validation {
+    condition     = var.blob_features.restore_policy_days == 0 || (var.blob_features.restore_policy_days >= 1 && var.blob_features.restore_policy_days <= 365)
+    error_message = "Restore policy days must be 0 to disable the policy or between 1 and 365."
+  }
+}
+
+variable "network_rules" {
```
Comment (Krusty93) [inline infra/modules/azure_storage_account/variables.tf:118]: Not sure if introducing this variable is valuable or not.
I mean, the storage account provides peps already, why does anyone need to stick to firewall rules?

**Review by Krusty93 — APPROVED** [review]

File: infra/modules/azure_storage_account/variables.tf (hunk 3 above)
Comment (christian-calabrese) [reply infra/modules/azure_storage_account/variables.tf:118]: I agree, in normal cases it shouldn't be needed. I added it for retro compatibility for storage accounts that need to enable public access.
What do you think?

File: infra/modules/azure_storage_account/alerts.tf (hunk 1 above)
Comment (christian-calabrese) [reply infra/modules/azure_storage_account/alerts.tf:7]: I wasn't sure actually, we can surely stick to the resource name that already contains all the information

File: infra/modules/azure_storage_account/variables.tf (hunk 3 above)
Comment (Krusty93) [reply infra/modules/azure_storage_account/variables.tf:118]: ok, for me it can stay

**Pull request merged**